pkg/liqoctl/move/movevolume_test.go

```diff
@@ -0,0 +1,614 @@
+				Expect(podSpec.RestartPolicy).To(Equal(corev1.RestartPolicyOnFailure))
+
+				Expect(podSpec.Containers).To(HaveLen(1))
+				Expect(podSpec.Containers[0].Image).To(Equal(resticImage))
+				Expect(podSpec.Containers[0].Args).To(Equal([]string{
+					"-r",
+					fmt.Sprintf("%s%s", resticRepositoryURL, oPvc.GetUID()),
+					"restore", "latest",
+					"--target", "/restore",
+				}))
+				Expect(podSpec.Containers[0].Env).To(ContainElement(corev1.EnvVar{
+					Name:  "RESTIC_PASSWORD",
+					Value: resticPassword,
+				}))
+				Expect(podSpec.Containers[0].VolumeMounts).To(HaveLen(1))
+				Expect(podSpec.Containers[0].VolumeMounts).To(ContainElement(corev1.VolumeMount{
+					Name:      "restore",
+					MountPath: "/restore",
+				}))
+
+				Expect(podSpec.Affinity).To(PointTo(Equal(corev1.Affinity{
+					NodeAffinity: &corev1.NodeAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+							NodeSelectorTerms: []corev1.NodeSelectorTerm{
+								{
+									MatchExpressions: []corev1.NodeSelectorRequirement{
+										{
+											Key:      "kubernetes.io/hostname",
+											Operator: corev1.NodeSelectorOpIn,
+											Values:   []string{nodeName},
+										},
+									},
+								},
+							},
+						},
+					},
+				})))
+
+				Expect(podSpec.Volumes).To(HaveLen(1))
+				Expect(podSpec.Volumes).To(ContainElement(corev1.Volume{
+					Name: "restore",
+					VolumeSource: corev1.VolumeSource{
+						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+							ClaimName: nPvc.GetName(),
+						},
+					},
+				}))
+			})
+
+		})
+
+	})
+
+	Context("ensure restic repository", func() {
+
+		var (
+			cl client.Client
+		)
+
+		Context("setup", func() {
+
+			var (
+				targetPvc *corev1.PersistentVolumeClaim
+			)
+
+			BeforeEach(func() {
+				cl = fake.NewClientBuilder().Build()
+
+				targetPvc = newPvc("pvc1")
+				targetPvc.Spec.Resources = corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("1Gi"),
+					},
+				}
+			})
+
+			It("creates a restic repository", func() {
+				Expect(ensureResticRepository(ctx, cl, targetPvc)).To(Succeed())
+
+				var repo appsv1.StatefulSet
+				Expect(cl.Get(ctx, types.NamespacedName{Name: resticRegistry, Namespace: liqoStorageNamespace}, &repo)).To(Succeed())
+
+				var svc corev1.Service
+				Expect(cl.Get(ctx, types.NamespacedName{Name: resticRegistry, Namespace: liqoStorageNamespace}, &svc)).To(Succeed())
+			})
+
+		})
+
+		Context("teardown", func() {
+
+			type deleteResticRepositoryTestcase struct {
+				client client.Client
+			}
+
+			DescribeTable("delete restic repository", func(c deleteResticRepositoryTestcase) {
+				Expect(deleteResticRepository(ctx, c.client)).To(Succeed())
+
+				var svc corev1.Service
+				Eventually(func() error {
+					return c.client.Get(ctx, types.NamespacedName{Name: resticRegistry, Namespace: liqoStorageNamespace}, &svc)
+				}).ShouldNot(Succeed())
```

Comment:
Could use the BeNotFound() error (in the liqo testutil package) to be more explicit about the error type (same for the other occurrences).